session/refresh.go

```diff
@@ -0,0 +1,80 @@
+package session
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/g8rswimmer/go-sfdc"
+)
+
+// Refresher is a structure will allow for a session token to be
+// re-authenticated based on a time interval.
+type Refresher struct {
+	session     *Session
+	sessionLock sync.RWMutex
+	err         error
+	stop        chan struct{}
+}
+
+// OpenRefresh will open the first session and set up a re-occuring refresh of the session token.
+func OpenRefresh(config sfdc.Configuration, refershTime int) (*Refresher, error) {
+	if refershTime <= 0 {
+		return nil, errors.New("session refresh: refresh time can not be less than zero")
```

Comment:
this error message technically doesn't align with the logic - is `< 1` sufficient?